### PR TITLE
add news collector

### PR DESCRIPTION
**Pull request by kaiyim**
This PR add news collector to get data from webhose. 

#120 
@JavierJia

### CODE REVIEW

File: noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/NewsCollection.java [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URLEncodedUtils;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Properties;
+
+
+public class NewsCollection {
```
Comment (JavierJia) [inline noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/NewsCollection.java]: I think it is better to name it as `WebHostCollector`.

File: noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/NewsCollection.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+    private static String apiKey = "";
+    private static String filePath = "";
+    private static String ts = null;
+    private static String propFileName = System.getProperty("user.dir") + "/noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/news.properties";
```
Comment (JavierJia) [inline noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/NewsCollection.java]: you should access it by the resource path other than the system path. @aishwaryakapse  knows how to access it.

File: noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/NewsCollection.java [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+            return;
+        }
+        while (true){
+
+            response_str += responseToString(response) + "\n";
+
+            if(response.moreResultsAvailable != 0) {
+                response = getMore(response);
+            }
+            else {
```
Comment (JavierJia) [inline noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/NewsCollection.java]: we are using the following format for `if else` statement
```
if {
} else {
}
```

File: noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/news.properties [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#Thu Jul 14 15:59:24 PDT 2016
```
Comment (JavierJia) [inline noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/news.properties]: this file should goes to `resource` folder.

Comment (JavierJia) [inline noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/news.properties]: it should be `webhose.config` to consist with the other config files.

File: noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/news.properties [hunk 5]
```diff
@@ -0,0 +1,4 @@
...
+filePath=/Users/KyleMa/Documents/workspace/cloudberry
```
Comment (JavierJia) [inline noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/news.properties]: default `filePath` should be `.`

File: noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/news.properties [hunk 6]
```diff
@@ -0,0 +1,4 @@
...
+ts=1468536911354
+apiKey=d29db614-cfaa-456d-a78b-397632cf1dd1
```
Comment (JavierJia) [inline noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/news.properties]: I think these `apiKey` should not be exposed publicly to the codebase.
You can just put the `input-your-webhose-key-here` to let us register and run it later.

File: noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/NewsCollection.java [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+                    output.close();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+
+        }
+    }
+
+    private static WebhoseResponse search_ts(String query, Long ts) throws IOException {
```
Comment (JavierJia) [inline noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/NewsCollection.java]: Use camelCase for function names.

Comment (JavierJia) [inline noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/NewsCollection.java]: rename `ts` to `fromTimeStamp`.

File: noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/news.properties (hunk 4 above)
Comment (JavierJia) [inline noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/news.properties]: Try to maintain the `properties` file is too heavy. I'd prefer to just put these three arguments through the command line. Then we can create a script and let the script handle the external logics.

**Comment by JavierJia** [comment]
looks OK. a few comments:
- [ ] pass the argument through command line instead of `property` file
- [ ] create a `webhose.sh` to run the `main` method.

File: noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/WebhoseCollector.java [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+            aliases = "--output",
+            usage = "output file directory")
+    private String filePath;
+
+    @Option(required = true,
+            name = "-t",
+            aliases = "--timestamp",
+            usage="timestamp for search")
+    private String ts;
+//    private static String propFileName = "";
```
Comment (JavierJia) [inline noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/WebhoseCollector.java]: remove unused lines.

File: noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/WebhoseCollector.java [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+//                    savePropValue();
+                } catch (IOException io) {
+                    io.printStackTrace();
+                }
+                break;
+            }
+        }
+    }
+
+//    private static void loadPropValue() {
```
Comment (JavierJia) [inline noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/WebhoseCollector.java]: remove unused codes.

File: noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/WebhoseCollector.java [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+    public static void main( String[] args ) throws IOException {
+        new WebhoseCollector().doMain(args);
+    }
+
+    public void doMain(String[] args) throws IOException {
+        CmdLineParser parser = new CmdLineParser(this);
+        try{
+            parser.parseArgument(args);
+        } catch( CmdLineException e ) {
+            // if there's a problem in the command line,
```
Comment (JavierJia) [inline noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/WebhoseCollector.java]: if the code is very clear about its functionality, we will not need this comment.

File: noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/WebhoseCollector.java [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+            // you'll get this exception. this will report
+            // an error message.
+            System.err.println(e.getMessage());
+            return;
+        }
+//        loadPropValue();
```
Comment (JavierJia) [inline noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/WebhoseCollector.java]: remove unused lines.

File: noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/WebhoseCollector.java [hunk 12]
```diff
@@ -0,0 +1,201 @@
...
+                //Update the timestamp
+                List<NameValuePair> params =
+                        URLEncodedUtils.parse(response.next, StandardCharsets.UTF_8);
+                for (final NameValuePair param : params) {
+                    if (param.getName().equals("ts")) {
+                        ts = param.getValue();
+                    }
+                }
+
+                File outFile = new File(filePath + "/response_" + ts + ".json");
```
Comment (JavierJia) [inline noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/WebhoseCollector.java]: Could we rename the outfile's name to `response_from_to.json`? The current one is only record the end ts right?

Comment (kaiyim) [inline noah/src/main/scala/edu/uci/ics/cloudberry/noah/news/WebhoseCollector.java]: Yeah, sure. I'll remove the unused lines and change the file name's format

**Comment by JavierJia** [comment]
👍

**Comment by JavierJia** [comment]
@kaiyim you can go ahead and merge.

**Pull request merged**